[BUG] Scala match error

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
```java
Caused by: scala.MatchError: BooleanType (of class org.apache.spark.sql.types.BooleanType$)
	at org.apache.spark.sql.execution.TiConverter$.fromSparkType(TiConverter.scala:29)
	at org.apache.spark.sql.catalyst.expressions.BasicExpression$.convertToTiExpr(BasicExpression.scala:85)
	at org.apache.spark.sql.catalyst.expressions.TiExprUtils$.isSupportedBasicExpression(TiExprUtils.scala:191)
	at org.apache.spark.sql.catalyst.expressions.TiExprUtils$.isSupportedFilter(TiExprUtils.scala:241)
	at org.apache.spark.sql.TiStrategy$$anonfun$12.apply(TiStrategy.scala:406)
	at org.apache.spark.sql.TiStrategy$$anonfun$12.apply(TiStrategy.scala:405)
```

**What did you do**
<!-- 
If possible, please provide a code receipt to produce this issue.
e.g., 
1. run `spark-shell --jars xxx.jar`
2. in spark-shell, run 
```
val df = spark.sql(“select sum(b) from t where a = 1”)
df.show(20)
```
3. update some data through TiDB
4. in spark-shell, run
```
val df = spark.sql(“select sum(b) from t where a = 1”)
df.show(20)
```
 -->
```mysql
mysql> CREATE TABLE `t` (
  `id` varchar(11) DEFAULT NULL,
  `flag` boolean DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
```

```scala
spark> spark.sql("SELECT * FROM t WHERE FLAG = 'true' and SUBSTR(ID, 1, 2) = '1'").show()
```

**Spark and TiSpark version info**
<!-- What version of Spark and TiSpark are you using? (Provide Spark version and run `spark.sql(“select ti_version()”).show(false)` in spark-shell) -->

TiSpark master
TiDB v4.0.5

<!--
**Additional context**
Add any other context about the problem here.
You may also provide TiDB version here if it is related to the issue.
-->

--- a/core/src/main/scala/org/apache/spark/sql/execution/TiConverter.scala
+++ b/core/src/main/scala/org/apache/spark/sql/execution/TiConverter.scala
@@ -27,6 +27,7 @@ object TiConverter {
     // TODO: review type system
     // pending: https://internal.pingcap.net/jira/browse/TISPARK-99
     tp match {
+      case _: sql.types.BooleanType => BitType.BIT
       case _: sql.types.BinaryType => BytesType.BLOB
       case _: sql.types.StringType => StringType.VARCHAR
       case _: sql.types.LongType => IntegerType.BIGINT

--- a/core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala
+++ b/core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala
@@ -21,6 +21,20 @@ import org.apache.spark.sql.functions.{col, sum}
 
 class IssueTestSuite extends BaseTiSparkTest {
 
+  test("Scala match error when predicate includes boolean type conversion") {
+    tidbStmt.execute("drop table if exists t")
+    tidbStmt.execute("""
+                       |CREATE TABLE `t` (
+                       |  `id` varchar(11) DEFAULT NULL,
+                       |  `flag` boolean DEFAULT NULL
+                       |) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+                       |""".stripMargin)
+
+    tidbStmt.execute("insert into t values('1', true), ('2', false), ('3', true), ('4', false)")
+
+    explainAndRunTest(s"SELECT * FROM t WHERE FLAG = 'true' and SUBSTR(ID, 1, 2) = '1'")
+  }
+
   test("large column number + double read") {
     tidbStmt.execute(
       resourceToString(

